src/components/Nav/Navigation.tsx

```diff
@@ -11,25 +11,60 @@ import ServiceGraphPage from '../../pages/ServiceGraph/ServiceGraphPage';
 import ServiceListPage from '../../pages/ServiceList/ServiceListPage';
 
 const istioRulesPath = '/rules';
-const istioRulesTitle = 'Istio Mixer';
+export const istioRulesTitle = 'Istio Mixer';
 const serviceGraphPath = '/service-graph/istio-system';
-const serviceGraphTitle = 'Graph';
+export const serviceGraphTitle = 'Graph';
 const servicesPath = '/services';
-const servicesTitle = 'Services';
+export const servicesTitle = 'Services';
 
 const pfLogo = require('../../img/logo-alt.svg');
 const pfBrand = require('../../assets/img/kiali-title.svg');
 
-class Navigation extends React.Component {
+type PropsType = {
+  location: any;
+};
+
+type StateType = {
+  selectedItem: string;
+};
+
+class Navigation extends React.Component<PropsType, StateType> {
   static contextTypes = {
     router: PropTypes.object
   };
 
   constructor(props: any) {
     super(props);
     this.navigateTo = this.navigateTo.bind(this);
+
+    // handle initial path from the browser
+    const selected = this.parseInitialPath(props.location.pathname);
+    this.state = {
+      selectedItem: `/${selected}/`
+    };
   }
 
+  parseInitialPath = (pathname: string) => {
+    let selected = '';
+    if (pathname.startsWith('/namespaces') || pathname.startsWith('/services')) {
```

Comment:
Minor. For consistency, I would suggest to use the constants for the path.
So, then, if some change, we have them located in a single point.
(And also to follow same style within same file).